File: gaia-sdk-python/gaia_sdk/graphql/request/input/UpdateRoleImpulse.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class UpdateRoleImpulse():
    """
    The specification to update a role instance
    """
    tenantId: str
    roleId: str
    name: str
    permissions: List[str]

    def __init__(self, tenantId: str, roleId: str, name: str, permissions: List[str]):
        self.tenantId = tenantId
        self.roleId = roleId
        self.name = name
        self.permissions = permissions

    def __eq__(self, other):
        if type(other) is type(self):
            return self.tenantId == other.tenantId and self.roleId == other.roleId and self.name == other.name and self.permissions == other.permissions
        return False

    def __repr__(self):
        return {'tenantId': self.tenantId, 'roleId': self.roleId, 'name': self.name, 'permissions': self.permissions}

File: gaia-sdk-javascript/src/graphql/request/input/UpdateRoleImpulse.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {RuntimeState} from "../enumeration/RuntimeState";
import {SkillState} from "../enumeration/SkillState";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * The specification to update a role instance
 */
export class UpdateRoleImpulse {
public _typeName = "UpdateRoleImpulse";
    private tenantId:Uuid;
    private roleId:Uuid;
    private name:String;
    private permissions:Array<String>;

    constructor (tenantId:Uuid, roleId:Uuid, name:String, permissions:Array<String>) {
        this.tenantId = tenantId;
        this.roleId = roleId;
        this.name = name;
        this.permissions = permissions;
    }
}
